src/ui/control-panel.ts

```diff
@@ -618,28 +533,34 @@ class ControlPanel extends Panel {
             if (e.ctrlKey || e.metaKey) {
                 // handle meta/ctrl keys
                 if (!e.shiftKey && e.key === 'z') {
-                    this.events.fire('undo');
+                    events.fire('edit:undo');
                 } else if (e.shiftKey && (e.key === 'z' || e.key === 'Z')) {
-                    this.events.fire('redo');
+                    events.fire('edit:redo');
                 }
             } else {
                 // handle non-meta/ctrl keys
                 if (e.key === 'Delete' || e.key === 'Backspace') {
-                    this.events.fire('deleteSelection');
+                    events.fire('deleteSelection');
                 } else if (e.key === 'Escape') {
-                    deactivate();
+                    events.fire('tool:activate', '');
+                } else if (e.key === '1') {
+                    events.fire('tool:activate', 'Move');
+                } else if (e.key === '2') {
+                    events.fire('tool:activate', 'Rotate');
+                } else if (e.key === '3') {
+                    events.fire('tool:activate', 'Scale');
                 } else if (e.key === 'R' || e.key === 'r') {
-                    toggle(boxSelection);
+                    events.fire('tool:activate', 'RectSelection');
                 } else if (e.key === 'F' || e.key === 'f') {
-                    this.events.fire('focusCamera');
+                    events.fire('focusCamera');
                 } else if (e.key === 'B' || e.key === 'b') {
-                    toggle(brushSelection);
+                    events.fire('tool:activate', 'BrushSelection');
                 } else if (e.key === 'I' || e.key === 'i') {
-                    this.events.fire('invertSelection');
+                    events.fire('invertSelection');
                 } else if (e.key === '[') {
-                    brushSelection.smaller();
+                    events.fire('BrushSelection:smaller');
```

Comment:
For some reason, I hate events with capital letters. Personal preference. 😄

Reply:
There were reasons, but I don't love it either. Made it start small.